src/simulation_strategies/CliffordsRSimulationStrategy.py

```diff
@@ -0,0 +1,44 @@
+__author__ = 'Tomasz Rybotycki'
+# One should note, that R is required to use this module, as original Clifford's program is written in R.
+# On my Windows 10, I am using anaconda and I had to add R_HOME env variable and R_path\bin, R_path\bin\x64 to the path.
+# https://cran.r-project.org/web/packages/BosonSampling/index.html
+
+from rpy2.robjects import packages
+from src.rpy2_utilities import numpy_array_to_r_matrix
+from numpy import array
+from src.simulation_strategies.SimulationStrategy import SimulationStrategy
+from src.Boson_Sampling_Utilities import particle_state_to_modes_state
+
+
+class CliffordsRSimulationStrategy(SimulationStrategy):
+    def __init__(self, number_of_bosons: int, interferometer_matrix: array) -> None:
+        self.interferometer_matrix = interferometer_matrix
+        self.number_of_bosons = number_of_bosons
+
+        required_packages = ('BosonSampling', 'Rcpp', 'RcppArmadillo')
```

Comment:
This part checking for installed packages, it would make an excellent utility. Have you considered moving it to some `utils`? It seems it could be reused multiple times ^_^

On the other hand, is this project using `pip freeze` to list all of the requirements? The usual flow is to do
```
pip freeze > requirements.txt
```
and then on another machine do
```
pip install -r requirements.txt
```

This could be not viable in this case because you might be using some particular conda, anaconda, miniconda or other version, but the "pure" python projects tend to start with this template.

Reply:
This check is for R packages only. I don't expect to use more R resources for now but in the future, if I change my mind, I'll consider it.